File: Java/Lista_03(Estrutura_Repetitiva)/Validacao_de_Nota.java
import java.util.Locale;
import java.util.Scanner;

public class Validacao_de_Nota {

	public static void main(String[] args) {

		Locale.setDefault(Locale.US);
		Scanner s = new Scanner(System.in);
		
		System.out.print("Digite a primeira nota: ");
		double nota1 = s.nextDouble();
		
		while(nota1<0 || nota1>10) {
			System.out.print("Valor invalido! Tente novamente: ");
			nota1 = s.nextDouble();
		}
		
		System.out.print("Digite a segunda nota: ");
		double nota2 = s.nextDouble();
		
		while(nota2<0 || nota2>10) {
			System.out.print("Valor invalido! Tente novamente: ");
			nota2 = s.nextDouble();
		}
		
		double media = (nota1+nota2)/2;
		
		System.out.printf("MEDIA = %.2f", media);
		
		s.close();
		
	}

}

File: C++/Lista_03(Estrutura_Repetitiva)/Validacao_de_Nota.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main(){

    double nota1, nota2, media;

    cout << "Digite a primeira nota: ";
    cin >> nota1;

    while (nota1<0.0 || nota1>10.0){
        cout << "Valor invalido! Tente novamente: ";
        cin >> nota1;
    }

    cout << "Digite a segunda nota: ";
    cin >> nota2;

    while (nota2<0.0 || nota2>10.0){
        cout << "Valor invalido! Tente novamente: ";
        cin >> nota2;
    }

    media = (nota1+nota2)/2;

    cout << "MEDIA = " << fixed << setprecision(2) << media;

    return 0;

}